### PR TITLE
[Multiple Datasource] Add Empty state no_data_source

### PR DESCRIPTION
**Pull request by yujin-emma**
### Description

Add the empty state  component for multiple data source, and add it on examples.
When enable multiple datasource feature, this empty state will display when there is no remote data source and hide the local cluster.

### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6420

## Screenshot

when there is no remote data source but do not hide local cluster

empty-state-lc

when there is no remote data source but do hide local cluster

empty-state-hide-lc

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/data_source_management/public/index.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import { from } from 'rxjs';
```
Comment (BionIT) [inline src/plugins/data_source_management/public/index.ts:6]: is this used?

File: src/plugins/data_source_management/public/components/no_data_source/no_data_source.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+
+import { EuiButtonEmpty } from '@elastic/eui';
+
+export const NoDataSource = () => {
+  const label = '    No data sources';
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/no_data_source/no_data_source.tsx:11]: why we have empty spaces here?

File: src/plugins/data_source_management/public/components/data_source_menu/types.ts [hunk 3]
```diff
@@ -39,6 +39,7 @@ export const DataSourceComponentType = {
   DataSourceView: 'DataSourceView',
   DataSourceAggregatedView: 'DataSourceAggregatedView',
   DataSourceMultiSelectable: 'DataSourceMultiSelectable',
+  NoDataSource: 'NoDataSource',
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/data_source_menu/types.ts:42]: why exposing it here? Can we mount this component when there is no data sources?

File: src/plugins/data_source_management/public/components/data_source_menu/data_source_menu.tsx [hunk 4]
```diff
@@ -108,6 +115,8 @@ export function DataSourceMenu<T>(props: DataSourceMenuProps<T>): ReactElement |
         return renderDataSourceView(componentConfig as DataSourceViewConfig);
       case DataSourceComponentType.DataSourceMultiSelectable:
         return renderDataSourceMultiSelectable(componentConfig as DataSourceMultiSelectableConfig);
+      case DataSourceComponentType.NoDataSource:
+        return renderNoDataSource(componentConfig as DataSourceBaseConfig);
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/data_source_menu/data_source_menu.tsx:119]: This is an internal component that we render when there is no data sources

File: src/plugins/data_source_management/public/components/no_data_source/no_data_source.tsx (hunk 2 above)
Comment (yujin-emma) [reply src/plugins/data_source_management/public/components/no_data_source/no_data_source.tsx:11]: removed

File: src/plugins/data_source_management/public/index.ts (hunk 1 above)
Comment (yujin-emma) [reply src/plugins/data_source_management/public/index.ts]: Not used, already removed

File: src/plugins/data_source_management/public/components/data_source_menu/data_source_menu.tsx (hunk 4 above)
Comment (yujin-emma) [reply src/plugins/data_source_management/public/components/data_source_menu/data_source_menu.tsx:119]: will remove

File: src/plugins/data_source_management/public/components/data_source_menu/types.ts (hunk 3 above)
Comment (yujin-emma) [reply src/plugins/data_source_management/public/components/data_source_menu/types.ts:47]: removed

File: src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.tsx [hunk 5]
```diff
@@ -63,6 +66,11 @@ export class DataSourceAggregatedView extends React.Component<
     this._isMounted = true;
     getDataSourcesWithFields(this.props.savedObjectsClient, ['id', 'title', 'auth.type'])
       .then((fetchedDataSources) => {
+        if (fetchedDataSources?.length === 0) {
+          this.setState({
+            showEmptyState: true,
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.tsx:71]: is it possible that we have local cluster?

File: src/plugins/data_source_management/public/components/data_source_menu/data_source_menu.test.tsx [hunk 6]
```diff
@@ -153,4 +153,16 @@ describe('DataSourceMenu', () => {
     );
     expect(container).toMatchSnapshot();
   });
+
+  it('should render no data source', () => {
+    component = shallow(
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/data_source_menu/data_source_menu.test.tsx:158]: let's not expose this component from data source menu

File: src/plugins/data_source_management/public/components/data_source_menu/data_source_menu.tsx [hunk 7]
```diff
@@ -80,7 +81,6 @@ export function DataSourceMenu<T>(props: DataSourceMenuProps<T>): ReactElement |
         hideLocalCluster={hideLocalCluster || false}
         fullWidth={fullWidth}
         uiSettings={uiSettings}
-        application={application}
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/data_source_menu/data_source_menu.tsx:83]: why removing application here?

File: src/plugins/data_source_management/public/components/data_source_multi_selectable/data_source_multi_selectable.tsx [hunk 8]
```diff
@@ -80,6 +83,7 @@ export class DataSourceMultiSelectable extends React.Component<
         ...this.state,
         selectedOptions,
         defaultDataSource,
+        showEmptyState: fetchedDataSources?.length === 0 || false,
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/data_source_multi_selectable/data_source_multi_selectable.tsx:86]: how about local cluster?

File: src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.tsx [hunk 9]
```diff
@@ -172,6 +178,9 @@ export class DataSourceSelectable extends React.Component<
         'title',
         'auth.type',
       ]);
+      if (fetchedDataSources?.length === 0) {
+        this.setState({ showEmptyState: true });
```
Comment (BionIT) [inline src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.tsx:182]: let's handle local cluster as well

File: src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.tsx (hunk 5 above)
Comment (yujin-emma) [reply src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.tsx:71]: updated

File: src/plugins/data_source_management/public/components/data_source_menu/data_source_menu.test.tsx (hunk 6 above)
Comment (yujin-emma) [reply src/plugins/data_source_management/public/components/data_source_menu/data_source_menu.test.tsx:158]: removed

File: src/plugins/data_source_management/public/components/data_source_multi_selectable/data_source_multi_selectable.tsx (hunk 8 above)
Comment (yujin-emma) [reply src/plugins/data_source_management/public/components/data_source_multi_selectable/data_source_multi_selectable.tsx:86]: updated

File: src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.tsx (hunk 9 above)
Comment (yujin-emma) [reply src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.tsx:182]: updated

File: src/plugins/data_source_management/public/components/data_source_menu/data_source_menu.tsx (hunk 7 above)
Comment (BionIT) [reply src/plugins/data_source_management/public/components/data_source_menu/data_source_menu.tsx:83]: can you check the main branch https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data_source_management/public/components/data_source_menu/data_source_menu.tsx and make sure we are not deleting existing implementations? cc @zhongnansu

File: src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.tsx (hunk 5 above)
Comment (BionIT) [reply src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.tsx:73]: is it possible that there is data source but filtered out?

**Comment by BionIT** [comment]
Can you close this PR if not relevant any more @yujin-emma?

**Comment by kavilla** [comment]
> Can you close this PR if not relevant any more @yujin-emma?

+1, I see the original issue is closed.

**Pull request closed without merging**